frontend/packages/pipelines-plugin/src/components/quicksearch/PipelineQuickSearchVersionDropdown.tsx

```diff
@@ -27,13 +27,14 @@ const PipelineQuickSearchVersionDropdown: React.FC<PipelineQuickSearchVersionDro
     return null;
   }
   const versionItems = versions.reduce((acc, { version }) => {
-    acc[version.toString()] =
+    acc[
+      item.attributes.installed.length === 3 ? parseFloat(version.toString()) : version.toString()
+    ] =
```

Comment:
This looks magic. Can you add a comment (to the code) explaining the `=== 3` check and the `parseFloat` here so the next reader will understand it?